### PR TITLE
feat: migrate ExpandableCard component to shadcn

### PR DESCRIPTION
**Pull request by kushagrasarathe**
Migrates `ExpandableCard` component to ShadCN from Chakra

## Description
This PR updates the ExpandableCard to use ShadcnUI's accordion component instead of Chakra component. Styles have been updated for both light and dark version.

## Note
@pettinarip I've used the `accordion` from `/tailwind` directory and did not directly overwrite the classes in the primitives, instead added classes in the `ExpandableCard` component to keep the UI similar to how it was with Chakra.

Lemme know if this approach doesn't look good, happy to update it. 

Also the default behaviour on shadcn accordion doesn't allow the accordion it to be closed if clicked on content, I can add this using mouse events, thoughts?

## Related Issue
#13946

### CODE REVIEW

File: src/components/ExpandableCard.tsx [hunk 1]
```diff
@@ -54,96 +52,40 @@ const ExpandableCard = ({
...
-          <AccordionButton
-            width="100%"
-            px="6"
-            py="6"
-            flex="1"
+    <>
+      <Accordion type="single" collapsible className="mb-4">
+        <AccordionItem
+          value="item-1"
+          className="rounded-sm border border-gray-200/50 hover:bg-gray-100/50 dark:border-gray-600 dark:hover:bg-gray-800"
```
Comment (pettinarip) [inline src/components/ExpandableCard.tsx:59]: A few color changes to match the new colors we have in the Design System
```suggestion
          className="rounded-sm border hover:bg-background-highlight"
```

File: src/components/ExpandableCard.tsx [hunk 2]
```diff
@@ -54,96 +52,40 @@ const ExpandableCard = ({
...
+          <AccordionTrigger
+            hideIcon
             onClick={onClick}
-            _hover={{
-              backgroundColor: "ednBackground",
-            }}
-            _expanded={{
-              backgroundColor: "transparent",
-            }}
+            className="w-full p-6 transition-colors hover:bg-gray-50 hover:text-black md:p-6 dark:hover:bg-gray-800 dark:hover:text-white [&[data-state=open]]:bg-transparent [&[data-state=open]]:text-black dark:[&[data-state=open]]:text-white"
```
Comment (pettinarip) [inline src/components/ExpandableCard.tsx:64]: Similar case with colors, lets simplify this by using some of the DS colors.
```suggestion
            className="w-full p-6 transition-colors hover:bg-background-highlight hover:text-black md:p-6 dark:hover:text-white [&[data-state=open]]:bg-transparent [&[data-state=open]]:text-black dark:[&[data-state=open]]:text-white"
```

File: src/components/ExpandableCard.tsx [hunk 3]
```diff
@@ -54,96 +52,40 @@ const ExpandableCard = ({
...
-                  marginBottom="0"
-                  width="fit-content"
-                >
+            <Flex className="w-full flex-col items-center text-left sm:flex-row">
+              <VStack className="items-center">
+                <HStack className="mb-2 mt-4">
+                  {Svg && <Svg className="mr-6" />}
+                  <h3 className="text-xl font-semibold">{title}</h3>
+                </HStack>
+                <p className="mb-0 w-fit text-sm text-gray-600">
```
Comment (pettinarip) [inline src/components/ExpandableCard.tsx:72]: color change and unnecessary margin class in this case
```suggestion
                <p className="w-fit text-sm text-body-medium">
```

File: src/components/ExpandableCard.tsx [hunk 4]
```diff
@@ -54,96 +52,40 @@ const ExpandableCard = ({
...
-              <Text
-                fontSize="md"
-                color="primary.base"
-                ms={{ base: undefined, sm: "auto" }}
-                mt="auto"
-                mb="auto"
-              >
+                </p>
+              </VStack>
+              <span className="my-auto text-md text-purple-600 sm:ml-auto dark:text-purple-400">
```
Comment (pettinarip) [inline src/components/ExpandableCard.tsx:76]:
```suggestion
              <span className="my-auto text-md text-primary sm:ml-auto">
```

File: src/components/ExpandableCard.tsx [hunk 5]
```diff
@@ -54,96 +52,40 @@ const ExpandableCard = ({
...
-          >
-            {children}
-          </Box>
-        </AccordionPanel>
-      </AccordionItem>
-    </Accordion>
+              </span>
+            </Flex>
+          </AccordionTrigger>
+          <AccordionContent className="cursor-pointer p-6 pt-0 md:p-6 md:pt-0">
```
Comment (pettinarip) [inline src/components/ExpandableCard.tsx:81]: Lets remove the cursor pointer now that we do not support that.
```suggestion
          <AccordionContent className="p-6 pt-0 md:p-6 md:pt-0">
```

File: src/components/ExpandableCard.tsx [hunk 6]
```diff
@@ -54,96 +52,40 @@ const ExpandableCard = ({
...
+            <div className="border-t border-gray-200 pt-6 text-md text-gray-800 dark:border-gray-600 dark:text-white">
```
Comment (pettinarip) [inline src/components/ExpandableCard.tsx:82]: Lets go with our default color on border and body color.
```suggestion
            <div className="border-t pt-6 text-md text-body">
```

**Review by pettinarip — CHANGES_REQUESTED** [review]
Great start @kushagrasarathe! The approach you took is perfect. Left a few requests on color token changes to match the Design System.

This is how I see it on https://deploy-preview-14017--ethereumorg.netlify.app/en/contributing/style-guide/
image

We would also need to align the text on the header texts to the left.

**Comment by kushagrasarathe** [comment]
Hey @pettinarip pushed the changes requestd 🙌 

> Great start @kushagrasarathe! The approach you took is perfect. Left a few requests on color token changes to match the Design System.
> 
> This is how I see it on https://deploy-preview-14017--ethereumorg.netlify.app/en/contributing/style-guide/ image
> 
> We would also need to align the text on the header texts to the left.

**Review by pettinarip — APPROVED** [review]
Thanks @kushagrasarathe !

**Pull request merged**